cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java

```diff
@@ -0,0 +1,191 @@
+        handler = new FetchFromIdentityNumberHandler(apiClient, environment);
+    }
+
+    @Test
+    void shouldReturnPersonWhenMockedHttpResponseContainsCristinPerson() throws Exception {
+        Person actual = sendQuery(defaultBody(), EMPTY_MAP).getBodyObject(Person.class);
+        String expectedString = IoUtils.stringFromResources(Path.of(NVA_API_GET_PERSON_RESPONSE_JSON));
+        Person expected = OBJECT_MAPPER.readValue(expectedString, Person.class);
+
+        assertThat(actual, equalTo(expected));
+    }
+
+    @Test
+    void shouldProduceCorrectCristinUriFromNationalIdentifier() throws IOException, ApiGatewayException {
+        apiClient = spy(apiClient);
+        handler = new FetchFromIdentityNumberHandler(apiClient, environment);
+        sendQuery(defaultBody(), EMPTY_MAP);
+
+        verify(apiClient).fetchQueryResults(new UriWrapper(VALID_CRISTIN_NATIONAL_ID_URI).getUri());
+        verify(apiClient).fetchGetResult(new UriWrapper(URI_FIRST_HIT_FROM_CRISTIN).getUri());
+    }
+
+    @Test
+    void shouldReturnServerErrorWhenBackendAuthenticationNotSentToUpstreamOrNotValid() throws Exception {
+        HttpClient clientMock = mock(HttpClient.class);
+        when(clientMock.<String>send(any(), any())).thenReturn(new HttpResponseFaker(EMPTY_STRING, 401));
+        CristinPersonApiClient apiClient = new CristinPersonApiClient(clientMock);
+        handler = new FetchFromIdentityNumberHandler(apiClient, environment);
+        GatewayResponse<Person> gatewayResponse = sendQuery(defaultBody(), EMPTY_MAP);
+
+        assertEquals(HttpURLConnection.HTTP_INTERNAL_ERROR, gatewayResponse.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON.toString(), gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_SERVER_ERROR));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenPayloadNotMatchingContract() throws Exception {
+        GatewayResponse<Person> gatewayResponse = sendInvalidQuery();
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON.toString(), gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_INVALID_PAYLOAD));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenSupplyingAnyQueryParam() throws Exception {
+        GatewayResponse<Person> gatewayResponse = sendQuery(defaultBody(), SOME_QUERY_PARAM);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON.toString(), gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertThat(gatewayResponse.getBody(), containsString(ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_PERSON_LOOKUP));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenInvalidTypeOrValueSpecified() throws Exception {
```

Comment:
Maybe split this into two tests? I got an error doing exactly the same and had two split into two separate `GatewayResponse` objects instead of reusing the same instance